Vagrantfile

```diff
@@ -4,10 +4,10 @@
 
 Vagrant.configure(2) do |config|
 
-    config.vm.box = "centosah/7alpha"
-    config.vm.box_url = 'https://ci.centos.org/artifacts/sig-atomic/centos-continuous/images-alpha/cloud/latest/images/centos-atomic-host-7-vagrant-libvirt.box'
+    # TODO: switch to the Fedora 30 CoreOS vagrant box once available
```

Comment:
Hmm, yeah these docs need an overhaul. Given we're not likely to release to Fedora 29 anymore, we should instead link to coreos-assembler and http://artifacts.ci.centos.org/fedora-coreos/prod/builds/latest/. (I've completely switched over to f30 and targeting FCOS/Silverblue when hacking on rpm-ostree now).

Vagrant definitely has a nice workflow for getting started, and we'll probably eventually offer a Vagrant box (https://github.com/coreos/fedora-coreos-tracker/issues/144), though it's a shame making this a blocker. Hmm, how about we recommend `cosa run -d`? E.g. `qemu-img create -b $fcos-img my-img.qcow2` and then `cosa run -d my-img.qcow2 --persist`. We could probably add a switch there to make it do this.

Eventually `cosa` should learn more tricks to facilitate development (see e.g. https://github.com/coreos/coreos-assembler/issues/470 and https://github.com/projectatomic/rpm-ostree/issues/1788).

Reply:
https://github.com/coreos/coreos-assembler/pull/507

Probably needs more things to be useful for development though, like https://github.com/coreos/coreos-assembler/issues/371.

Reply:
Will try out the workflow with `cosa run -d` - so far I've only used the `overrides/rpm` method with COSA. Thank you for adding the `--persist` option in already, will give that a go as well!

IMO I think the vagrant docs would be handy to keep around for the time being, as an alternative for someone to base off to try with their own pet container. These edits would go out of date once F29 is EOL though. The FCOS box may be available then - but agreed that shouldn't be made a blocker for this. Recommending  `cosa` primarily SGTM - should definitely link to COSA-based docs first and refer contributors there.

I wonder if we could give steps for COSA in `HACKING.md`, so contributors are directed to that first. Then we don't link the vagrant docs from `HACKING.md`, but leave the `README.md` there?

Reply:
Hmm - reading this again it definitely makes sense to only link to a F30-based flow (given we're likely not releasing to F29 anymore).

I still like the idea of offering an alternative pattern using FCOS Vagrant boxes if they become available, but probably makes sense to do the overhaul of these docs now and add support to `cosa` as needed.

Reply:
Didn't mean to sign you up for reworking this whole thing! :) I'm cool with either holding this until we flesh out the `cosa` workflow/start producing Vagrant boxes or maybe for now just a header to say that these instructions are in flux and that development should target FCOS with a link to cosa and http://artifacts.ci.centos.org/fedora-coreos/prod/builds/latest/?

Speaking of, I guess it's time to update our CI again to reflect that...

Reply:
> f30 vagrant box is available now

well - f30 cloud base one is and it looks like you're using atomic host - so nevermind :)